js/snake.js

```diff
@@ -214,13 +217,31 @@ function start() {
     game = new Game(canvas.width, canvas.height);
     document.onkeydown = function (e) { game.keyDown(e); };
     var ctx = canvas.getContext("2d");
-    setInterval(loop, 75, game, ctx);
+    loop(game, ctx);
 }
 
 // The game update and rendering loop
 function loop(game, ctx) {
     game.update();
     game.render(ctx);
+	clearInterval(timer);
+	// every 5 food eaten, increase spead by 10 ms
+	var snakeFastIndex = Math.floor(game.snake.length / 5);
+	console.log("snakeFastIndex = "+snakeFastIndex);
+	if(snakeFastIndex == 0){
+		var speed = 80;
+	}
+	else{
+		// if speed should be faster to 30 ms, keep it to 30 ms, so the player can still play
+		if(snakeFastIndex>5){
+			var speed = 30;
+		}
+		else{
+			var speed = 80 - (10*snakeFastIndex);
+		}
+	}
+	console.log("snake speed = "+speed);
+	timer = setInterval(loop, speed, game, ctx);

```

Comment:
I think you want to use `setTimeout` here instead of `setInterval`, don't you ?

Reply:
True. Will commit ne changes containing this